pkg/maps/ctmap/ctmap.go

```diff
@@ -618,52 +676,84 @@ func GetInterval(m *Map, stats gcStats) (interval time.Duration) {
 	return calculateInterval(m, interval, stats)
 }
 
-func calculateInterval(m *Map, prevInterval time.Duration, stats gcStats) (interval time.Duration) {
-	interval = prevInterval
-
-	maxDeleteRatio := float64(stats.Deleted) / float64(m.MapInfo.MaxEntries)
-	if maxDeleteRatio == 0.0 {
-		return
+func thresholdGC() float64 {
+	switch option.Config.ConntrackGCProfile {
+	case option.ConntrackGCProfileAggressive:
+		return 0.01
+	case option.ConntrackGCProfileNormal:
+		return 0.3
+	case option.ConntrackGCProfileLazy:
+		fallthrough
+	default:
+		return 0.6
 	}
+}
 
-	switch {
-	case maxDeleteRatio > 0.25:
-		if maxDeleteRatio > 0.9 {
-			maxDeleteRatio = 0.9
+func calculateInterval(m *Map, prevInterval time.Duration, stats gcStats) (interval time.Duration) {
+	interval = prevInterval
+	threshold := thresholdGC()
+	thresholdHit := false
+	sum := uint32(0)
```

Comment:
I dropped this aspect since "# deleted elements" reflects the current state, but does not tell anything about the remaining *alive* entries that we've walked which are classified into the DyingEntries[] time buckets, but potentially both could be combined, e.g. when we have a significant amount of traffic such that alive entries fall into lower DyingEntries[] buckets where they hit the profile threshold and define a next short wakeup and we find that on the next run they are all still alive, we could back off a bit to have less walks.

Reply:
Yeah, this stuff is notoriously difficult to validate and construct environments to prove that changes like this are improving the behaviour. :/ I wonder if there's any academic literature on the topic that could be of interest..